modules/request-quota-increase/main.tf

```diff
@@ -3,24 +3,149 @@ terraform {
 }
 
 resource "aws_servicequotas_service_quota" "increase_quotas" {
+  for_each = local.resources_code
+
+  quota_code   = each.value.quota_code
+  service_code = each.value.service_code
+  value        = each.value.value
+}
+
+data "aws_servicequotas_service_quota" "by_quota_name" {
   for_each = var.resources_to_increase
 
-  quota_code   = local.codes[each.key].quota_code
-  service_code = local.codes[each.key].service_code
-  value        = each.value
+  quota_name   = local.resource_names[each.key].quota_name
+  service_code = local.resource_names[each.key].service_code
 }
 
+data "aws_region" "current" {}
+
 locals {
-  # PRs to add more of these mappings are very welcome. For more information
-  # on how to find the Service Code and Quota Code, see the README.md!
-  codes = {
-    nacl_rules = {
-      quota_code   = "L-2AEEBF1A"
+
+  resources_code = {
+    for key, value in var.resources_to_increase :
+    key => {
+      quota_code   = data.aws_servicequotas_service_quota.by_quota_name[key]["quota_code"]
+      service_code = local.resource_names[key].service_code
+      value        = value
+    } if !(data.aws_region.current.name != "us-east-1" && substr(key, 0, 3) == "iam")
+    # It will ignore the resource if it's IAM and it is not on us-east-1 region
```

Comment:
NIT: Update comment to explain that IAM _only_ works in us-east-1, which is why you need to skip it in other regions.